lib/tasks/demo.rake

```diff
@@ -14,84 +14,129 @@ namespace :demo do
-        FactoryGirl.create(:offer, :reviewed, :with_demo_items, :with_messages, created_by: donor)
-        FactoryGirl.create(:offer, :scheduled, :with_transport, :with_demo_items, :with_messages, created_by: donor)
-        offer = FactoryGirl.create(:offer, :closed, :with_transport, :with_demo_items, :with_messages, created_by: donor)
-        Package.joins(item: :offer).where("offers.id = ?", offer.id).each do |package|
-          package.allow_web_publish = true
-        end
+        offer = create_submitted_offer
+        puts "\t\tCreated Offer #{offer.id} in 'submitted' state"
+        offer = create_reviewing_offer
+        puts "\t\tCreated Offer #{offer.id} in 'under_review' state"
+        offer = create_reviewed_offer
+        puts "\t\tCreated Offer #{offer.id} in 'reviewed' state"
+        offer = create_scheduled_offer
+        puts "\t\tCreated Offer #{offer.id} in 'scheduled' state"
+        offer = create_closed_offer
+        puts "\t\tCreated Offer #{offer.id} in 'closed' state"
+        offer = create_inactive_offer
+        puts "\t\tCreated Offer #{offer.id} in 'inactive' state"
+        offer = create_receiving_offer
+        puts "\t\tCreated Offer #{offer.id} in 'receiving' state"
+        offer = create_recieved_offer
+        puts "\t\tCreated Offer #{offer.id} in 'received' state"
       end
     end
 
-    def create_package
-      # Create Packages for Goodcity and Items for Stockit
-      puts "Package:\t\tCreating #{count} Packages for Goodcity and Items for Stockit(with_item, with_set_item, received(published and unpublished) & stockit_package"
-
-      count.times do
-        FactoryGirl.create(:package, :with_item, :package_with_locations, :with_inventory_number)
-      end
-      count.times do
-        FactoryGirl.create(:package, :stockit_package, :with_inventory_number , :package_with_locations)
-      end
-      count.times do
-        FactoryGirl.create(:package, :with_set_item, :package_with_locations)
-      end
+    def create_orders
+      puts "Orders:\tCreating #{count} designated and #{count} dispatched order with orders_packages "
       count.times do
-        FactoryGirl.create(:package, :received, :with_inventory_number)
+        create_designated_packages
+        puts "\t\tCreated Order #{Order.last.id} with packages in Designated State"
+        create_dispatched_packages
+        puts "\t\tCreated Order #{Order.last.id} with packages in Dispatched State"
       end
-      count.times do
-        FactoryGirl.create(:package, :received, :with_inventory_number, :published)
+    end
+
+    def create_submitted_offer
+      FactoryGirl.create(:offer, :with_demo_items, :with_messages, created_by: donor).tap(&:submit)
+    end
+
+    def create_reviewing_offer
+      User.current_user = reviewer
+      create_submitted_offer.tap(&:start_review)
+    end
+
+    def create_reviewed_offer
+      offer = create_reviewing_offer
+      offer.update(reviewed_by: reviewer)
```

Comment:
for offer.update_column will(reviewed_by: reviewer) 
[Validations are skipped.
Callbacks are skipped.
updated_at/updated_on are not updated.](https://cbabhusal.wordpress.com/2014/10/08/different-between-update-update_columns-update_column-update_attributes-assign_attributes/)

So i don't think we might want to use it.